python/mlc_chat/support/auto_target.py

```diff
@@ -44,6 +45,26 @@ def detect_target_and_host(target_hint: str, host_hint: str) -> Tuple[Target, Bu
     return target, build_func
 
 
+def detect_tvm_device(target: Target) -> Device:
+    """Detect the TVM device from the target device.
+
+    Parameters
+    ----------
+    target : Target
+        The target device.
+
+    Returns
+    -------
+    device : Device
+        The TVM device.
+    """
+    if target.kind.name == "llvm":
+        return cpu()
+    if target.kind.name == "cuda":
+        return cuda()
+    raise ValueError(f"Unsupported device type from {target}")
```

Comment:
We could create device from `--device` string, for example, `device("cpu")`

Reply:
That's a better solution here. I notcied in https://github.com/mlc-ai/mlc-llm/pull/1172#discussion_r1379334922 there's another way to detect device from the weight itself, I think that might work even better because we don't need to pass device all the way as an argument and it would be accurate from the weight's ndarray device.

Reply:
They are not contradictory approaches. In fact, the workflow is as follows
- Step 1. the command line tool creates `Target` and `Device` from string using the detection logic that @LeshengJin suggests
- Step 2. the loader moves the NDArray to the `Device` from Step 1;
- The quantizer recognize the target either using `Target.current()` if it exists, or `NDArray.device` as a fallback

Also CC @cyx-6